memcache/selector.go

```diff
@@ -42,8 +42,16 @@ type ServerSelector interface {
 	// This can be used to get information which is server agnostic.
 	PickAnyServer() (net.Addr, error)
 	Each(func(net.Addr) error) error
+
+	// SetServers updates the list of servers available for selection.
+	SetServers(servers ...string) error
```

Comment:
does this need to be thread-safe?

Reply:
Yup it does. The `ServerSelector` docs (above) state this:

> All ServerSelector implementations must be safe for concurrent use by multiple goroutines.

Reply:
ah, you're not implementing this yourself. otherwise i would have been confused by the absence of a mutex

Reply:
Yeah I am just wrapping an existing implementation. The modularity in this lib is nice.